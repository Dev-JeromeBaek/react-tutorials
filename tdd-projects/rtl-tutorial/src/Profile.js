import React from 'react';

function Profile({ username, name }) {
  return (
    <div>
      <b>{username}</b>
      <span>({name})</span>
      <span>new</span>
    </div>
  );
}

export default Profile;