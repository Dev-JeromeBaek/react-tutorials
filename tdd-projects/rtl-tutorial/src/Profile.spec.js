import React from 'react';
import { render } from '@testing-library/react';
import Profile from './Profile';

describe('<Profile />', () => {
  it('matches snapshot', () => {
    const utils = render(<Profile username="jeromeBaek" name="백승엽" />);
    expect(utils.container).toMatchSnapshot();
  });
  it('shows the props correcly', () => {
    const utils = render(<Profile username="jeromeBaek" name="백승엽" />);
    utils.getByText('jeromeBaek');  // jeromeBaek 이라는 텍스트를 가진 엘리먼트가 있는지 확인
    utils.getByText('(백승엽)');  // (백승엽) 이라는 텍스트를 가진 엘리먼트가 있는지 확인
    utils.getByText(/백/);  // 정규식 /백/ 을 통과하는 엘리먼트가 있는지 확인
  });
});