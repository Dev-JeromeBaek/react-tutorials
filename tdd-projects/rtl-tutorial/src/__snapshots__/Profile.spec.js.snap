// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Profile /> matches snapshot 1`] = `
<div>
  <div>
    <b>
      jeromeBaek
    </b>
    <span>
      (
      백승엽
      )
    </span>
  </div>
</div>
`;
